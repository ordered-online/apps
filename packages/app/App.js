import { AppLoading } from 'expo';
import { Asset } from 'expo-asset';
import * as Font from 'expo-font';
import React, { Component } from 'react';
import {
  Platform,
  StatusBar,
  StyleSheet,
  SafeAreaView,
  View,
} from 'react-native';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';
import { store, persistor } from './store';
import { Ionicons } from '@expo/vector-icons';

import AppNavigator from './navigation/AppNavigator';
import { LinearGradient } from 'expo-linear-gradient';

export default class App extends Component {
  state = {
    isLoadingComplete: false,
  };

  render() {
    const { isLoadingComplete } = this.state;
    if (!isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={loadResourcesAsync}
          onError={handleLoadingError}
          onFinish={() => this.setState({ isLoadingComplete: true })}
        />
      );
    } else {
      return (
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <LinearGradient
              style={{ width: '100%', height: '100%' }}
              colors={['#57c75e', '#27ae60']}>
              <View style={styles.container}>
                {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
                <View style={styles.roundedBackground}>
                  <AppNavigator />
                </View>
              </View>
            </LinearGradient>
          </PersistGate>
        </Provider>
      );
    }
  }
}

async function loadResourcesAsync() {
  await Promise.all([
    Asset.loadAsync([require('./assets/icon.png')]),
    Font.loadAsync({
      ...Ionicons.font,
    }),
  ]);
}

function handleLoadingError(error) {
  // In this case, you might want to report the error to your error reporting
  // service, for example Sentry
  console.warn(error);
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: 'flex',
    justifyContent: 'center',
    alignContent: 'center',
    flexDirection: 'column',
  },
  roundedBackground: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    marginLeft: 8,
    marginRight: 8,
    marginBottom: 8,
    backgroundColor: '#f8f8f8',
    borderBottomLeftRadius: 32,
    borderBottomRightRadius: 32,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.3,
    shadowRadius: 16,
    elevation: 0.5,
  },
});
