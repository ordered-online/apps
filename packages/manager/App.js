import { AppLoading } from 'expo';
import { Asset } from 'expo-asset';
import * as Font from 'expo-font';
import React, { Component } from 'react';
import { Platform, StatusBar, StyleSheet, SafeAreaView } from 'react-native';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';
import { store, persistor } from './store';
import { Ionicons } from '@expo/vector-icons';

import AppNavigator from './navigation/AppNavigator';

export default class App extends Component {
  state = {
    isLoadingComplete: false,
  };

  componentDidMount() {}

  render() {
    const { isLoadingComplete } = this.state;
    if (!isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={loadResourcesAsync}
          onError={handleLoadingError}
          onFinish={() => this.setState({ isLoadingComplete: true })}
        />
      );
    } else {
      return (
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <SafeAreaView style={styles.container}>
              {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
              <AppNavigator />
            </SafeAreaView>
          </PersistGate>
        </Provider>
      );
    }
  }
}

async function loadResourcesAsync() {
  await Promise.all([
    Asset.loadAsync([require('./assets/icon.png')]),
    Font.loadAsync({
      ...Ionicons.font,
    }),
  ]);
}

function handleLoadingError(error) {
  // In this case, you might want to report the error to your error reporting
  // service, for example Sentry
  // console.warn(error);
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: 'flex',
    justifyContent: 'center',
    alignContent: 'center',
    flexDirection: 'column',
  },
});
