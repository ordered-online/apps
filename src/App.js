import React, { Component } from 'react';
import { StyleSheet, Text, View, Platform } from 'react-native';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';
import { Router, Switch, Route, Link } from './routing';
import { createBrowserHistory } from 'history';
import { store, persistor } from './store';
import Home from './components/Home';

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <Router history={ Platform.OS === 'web' ? createBrowserHistory() : null}>
            <View style={styles.container}>
              <View style={styles.nav}>
                <Link to="/">
                  <Text>Home</Text>
                </Link>
              </View>

              {/* A <Switch> looks through its children <Route>s and
                renders the first one that matches the current URL. */}

              <Switch>
                <Route exact path="/" component={Home} />
              </Switch>
            </View>
          </Router>
        </PersistGate>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: 25,
    padding: 20,
  },
  nav: {
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
});
