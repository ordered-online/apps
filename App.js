import React, { Component } from 'react';
import { StyleSheet, SafeAreaView } from 'react-native';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';
import { store, persistor } from './src/store';
import { AppContainer } from './src/AppContainer';

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <SafeAreaView style={styles.container} >
            <AppContainer />
          </SafeAreaView>
        </PersistGate>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    justifyContent: "center",
    alignContent: "center",
    flexDirection: "column"
  }
});
