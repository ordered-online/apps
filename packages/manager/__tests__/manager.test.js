import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

jest.mock('../navigation/AppNavigator', () => 'AppNavigator');

jest.mock('redux-persist/integration/react', () => ({
  PersistGate: props => props.children,
}));

describe('Manager', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    jest.useFakeTimers();
  });

  it(`manager renders the loading screen`, () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`manager renders the root without loading screen`, () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
