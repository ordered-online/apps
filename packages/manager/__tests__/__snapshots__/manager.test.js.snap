// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manager manager renders the loading screen 1`] = `
<AppLoading
  onError={[Function]}
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;

exports[`Manager manager renders the root without loading screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignContent": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <AppNavigator />
</RCTSafeAreaView>
`;
